### ISSUE TITLE
Tests fail when nosetests is not nosetests3

### ISSUE THREAD

**Issue by mpe** [amboar/fpos#21]
eg:

```
$ make check
python3 setup.py check 
running check
nosetests
E
======================================================================
ERROR: Failure: ImportError (No module named matplotlib)
----------------------------------------------------------------------
...
Ran 1 test in 0.003s

FAILED (errors=1)
```

But if I install `python3-nose` (Ubuntu):

```
$ nosetests3
.....................................................................E.E.......EEEEEEEEEEE................................
...
Ran 122 tests in 0.292s
```

Which is better, although I am still getting some errors.

So maybe the `Makefile` should explicitly call `nosetests3`?

**Comment by amboar** [comment]
@mpe @shenki I've pushed some fixes to the [gh-21-wip](https://github.com/amboar/fpos/commits/gh-21-wip) branch - do you mind testing it? A couple of things there:

* I've added `nose` to `requirements.txt`, so it will be available after a `make pip` without resorting to the system package manager to install it
* `make check` tries to detect available nosetest executables for python3 (see 7ef8d1dd), and should pick up `nosetests3` if you have that installed (or e.g. `nosetests-3.4` if you've installed via pip)

There are some other tweaks applied as well, hopefully they don't make things worse!

Finally, @mpe I can't seem to reproduce the failures, can you provide the relevant output of `nosetests3 -v`?

**Comment by mpe** [comment]
I'm trying, but I've spent about 3 hours trying to get matplotlib to install via pip and haven't succeeded yet. Will persist.

**Comment by amboar** [comment]
:( @mpe yeah it does take some time. I added mention of libblas-dev, liblapack-dev and gfortran to the readme last night after suffering through the similar problems. Thanks for persisting!

**Comment by mpe** [comment]
Ah thanks, helps to read the documentation :)

Now I get:

```
$ make check
python3 setup.py check 
running check
$
```

Because I did `install-user` and `nosetests` is in `~/.local/bin/` which is not in my `PATH` !

If I fix that all the tests pass!

**Comment by amboar** [comment]
Okay, the path issue is probably worth mentioning in the readme as well, I'll fix that. I'll make the Makefile bail if we can't find a nosetests when running the check target

**Comment by amboar** [comment]
Fixed in master as of 62d24e90
